Makefile

```diff
@@ -185,121 +199,393 @@ test-integration: test-integration-aws test-integration-clickhouse test-integrat
+	$(CONTAINER_TOOL) run -d --$(CONTAINER_ENCLOSURE)=test-integration-aws --name ec2_metadata \
+	 timberiodev/mock-ec2-metadata:latest
+	$(CONTAINER_TOOL) run -d --$(CONTAINER_ENCLOSURE)=test-integration-aws --name localstack \
+	 -e SERVICES=kinesis:4568,s3:4572,cloudwatch:4582,elasticsearch:4571,firehose:4573 \
+	 localstack/localstack@sha256:f21f1fc770ee4bfd5012afdc902154c56b7fb18c14cf672de151b65569c8251e
+	$(CONTAINER_TOOL) run -d --$(CONTAINER_ENCLOSURE)=test-integration-aws --name mockwatchlogs \
+	 -e RUST_LOG=trace luciofranco/mockwatchlogs:latest
+else
+	$(CONTAINER_TOOL) $(CONTAINER_ENCLOSURE) create test-integration-aws
+	$(CONTAINER_TOOL) run -d --$(CONTAINER_ENCLOSURE)=test-integration-aws -p 8111:8111 --name ec2_metadata \
+	 timberiodev/mock-ec2-metadata:latest
+	$(CONTAINER_TOOL) run -d --$(CONTAINER_ENCLOSURE)=test-integration-aws -p 4568:4568 -p 4572:4572 \
+	 -p 4582:4582 -p 4571:4571 -p 4573:4573 --name localstack \
+	 -e SERVICES=kinesis:4568,s3:4572,cloudwatch:4582,elasticsearch:4571,firehose:4573 \
+	 localstack/localstack@sha256:f21f1fc770ee4bfd5012afdc902154c56b7fb18c14cf672de151b65569c8251e
+	$(CONTAINER_TOOL) run -d --$(CONTAINER_ENCLOSURE)=test-integration-aws -p 6000:6000 --name mockwatchlogs \
+	 -e RUST_LOG=trace luciofranco/mockwatchlogs:latest
+endif
+
+stop-integration-aws:
+	$(CONTAINER_TOOL) rm --force ec2_metadata mockwatchlogs localstack 2>/dev/null; true
+ifeq ($(CONTAINER_TOOL),podman)
+	$(CONTAINER_TOOL) $(CONTAINER_ENCLOSURE) stop --name=test-integration-aws 2>/dev/null; true
+	$(CONTAINER_TOOL) $(CONTAINER_ENCLOSURE) rm --force --name=test-integration-aws 2>/dev/null; true
+else
+	$(CONTAINER_TOOL) $(CONTAINER_ENCLOSURE) rm test-integration-aws 2>/dev/null; true
+endif
+
 test-integration-aws: ## Runs AWS integration tests
 ifeq ($(AUTOSPAWN), true)
-	${MAYBE_ENVIRONMENT_EXEC} $(CONTAINER_TOOL)-compose up -d dependencies-aws
-	sleep 5 # Many services are very lazy... Give them a sec...
+	$(MAKE) -k stop-integration-aws \
+    ; rc=$$? \
+	$(MAKE) start-integration-aws
+	sleep 5 # Many services are very slow... Give them a sec...
 endif
 	${MAYBE_ENVIRONMENT_EXEC} cargo test --no-default-features --features aws-integration-tests --lib ::aws_ -- --nocapture
 ifeq ($(AUTODESPAWN), true)
-	${MAYBE_ENVIRONMENT_EXEC} $(CONTAINER_TOOL)-compose stop
+	$(MAKE) -k stop-integration-aws
+endif
+
+start-integration-clickhouse:
+ifeq ($(CONTAINER_TOOL),podman)
+	$(CONTAINER_TOOL) $(CONTAINER_ENCLOSURE) create --replace --name test-integration-clickhouse -p 8123:8123
+	$(CONTAINER_TOOL) run -d --$(CONTAINER_ENCLOSURE)=test-integration-clickhouse --name clickhouse yandex/clickhouse-server:19
+else
+	$(CONTAINER_TOOL) $(CONTAINER_ENCLOSURE) create test-integration-clickhouse
+	$(CONTAINER_TOOL) run -d --$(CONTAINER_ENCLOSURE)=test-integration-clickhouse -p 8123:8123 --name clickhouse yandex/clickhouse-server:19
+endif
+
+stop-integration-clickhouse:
+	$(CONTAINER_TOOL) rm --force clickhouse 2>/dev/null; true
```

Comment:
I feel like some of these commands could be `@` prefixed to quite them down a bit.